kyaml/yaml/fns.go

```diff
@@ -725,7 +728,7 @@ func (s FieldSetter) Filter(rn *RNode) (*RNode, error) {
 	}
 
 	// Clear the field if it is empty, or explicitly null
```

Comment:
I was wondering why this change was necessarily.  I tried it out and it was because otherwise the nil we so carefully preserved would be cleared.  Can we update the comment though, maybe adding "...unless it is our KeepNull marker" (though admittedly this makes the IsTaggedNull / "explicitly null" sort of confusing)

Reply:
> I was wondering why this change was necessarily. I tried it out and it was because otherwise the nil we so carefully preserved would be cleared. Can we update the comment though, maybe adding "...unless it is our KeepNull marker" (though admittedly this makes the IsTaggedNull / "explicitly null" sort of confusing)

Good call, I tried to express the my understanding with [c18fcbc](https://github.com/kubernetes-sigs/kustomize/pull/5519/commits/c18fcbced6a098586dfc344ec248c1b88b897601)